src/Database/Redis/Cluster.hs

```diff
@@ -214,23 +216,31 @@ moved _ = False
 
 
 nodeConnWithHostAndPort :: ShardMap -> Connection -> Host -> Port -> Maybe NodeConnection
-nodeConnWithHostAndPort shardMap (Connection nodeConns _ _ _) host port = do
+nodeConnWithHostAndPort shardMap (Connection nodeConns _ _ _ _) host port = do
     node <- nodeWithHostAndPort shardMap host port
     HM.lookup (nodeId node) nodeConns
 
 nodeConnectionForCommand :: Connection -> ShardMap -> [B.ByteString] -> IO NodeConnection
-nodeConnectionForCommand (Connection nodeConns _ _ infoMap) (ShardMap shardMap) request = do
+nodeConnectionForCommand (Connection nodeConns _ _ infoMap connReadOnly) (ShardMap shardMap) request = do
     let mek = case request of
           ("MULTI" : key : _) -> Just [key]
           ("EXEC" : key : _) -> Just [key]
           _ -> Nothing
+        isCmdReadOnly = CMD.isCommandReadonly infoMap request
```

Comment:
move this function inside the case where it's being used.

Reply:
done